### ISSUE TITLE
Quality flags are missing from the TessQualityFlags class

### ISSUE THREAD

**Issue by mrtommyb** [lightkurve/lightkurve#1163]
#### Problem description

The TESS quality flags were updated. They are now 16 bit. They are described [here](https://archive.stsci.edu/files/live/sites/mast/files/home/missions-and-data/active-missions/tess/_documents/EXP-TESS-ARC-ICD-TM-0014-Rev-F.pdf), which was updated in Sep 2020.

We should add these flags to the TESSQualityFlags class. I think that it may be worth a discussion on whether any of the new flags get added to _default_ or _hard_, but they should be added to _hardest_.
![Screen Shot 2021-11-22 at 11 36 39 AM](https://user-images.githubusercontent.com/1900573/142899990-1c8f28eb-7635-4391-abf4-67e33186d152.png)

**Comment by mrtommyb** [comment]
@rebekah9969 do you have a feel if bits 14-16 compromise the data at all, and should be excluded by default. I suspect the answer is no, but I don't know for sure.

**Comment by barentsen** [comment]
@rebekah9969 Any update on whether bits 14-16 compromise the data?

I will go ahead and add these flags to the "hardest" mask only unless I hear otherwise.

**Comment by rebekah9969** [comment]
@barentsen and @mrtommyb I contacted Michael about this and he said the following...

"Bit 14 is specific to planet science, and different users may find the data entirely acceptable.  You probably don't need to exclude this bit by default, although we haven't used it since year 2 and so it may never make a difference.

Bit 15 and 16 are pretty serious--- for 15, there shouldn't be any valid data except in the RAW_CNTS TPF column, I think.  Best to throw these data out by default.

Bit 16 means that PDC cannot apply a correction for that time stamp.  There is SAP_FLUX data at that time that may be useful for some researchers, but there is no PDC data at these times.  I don't know if what the default assumptions for lightkurve would be in this case.  I'd always look at the difference between SAP and PDCSAP flux and decide on a case by case basis, so I wouldn't want this bit removed by default.  But I could see how it may confuse someone if SAP_FLUX and PDCSAP_FLUX have different lengths."

I would therefore exclude 15 at minimum.

**Comment by orionlee** [comment]
I'd suggest at the minimal we should add the flags to TESSQualityFlags class, and include them all in the `hardest` (unless we change its definition).

Anecdotally, there are noticeable amount of cadences that are flagged as such. 
The following are some examples from some 2-minute lightcurve files, loaded with default mask.

```
TIC   sector author #bit14 #bit15 #bit16 #cadence
416537915 14 QLP 218 0 0 1237
416537915 15 QLP 220 0 0 1190
416537915 21 QLP 164 0 0 1262
416537915 41 SPOC 0 0 0 18322
416537915 48 SPOC 1698 876 1166 18989

TIC   sector author #bit14 #bit15 #bit16 #cadence
198394082 14 QLP 285 0 0 1237
198394082 15 QLP 395 0 0 1190
198394082 16 QLP 120 0 0 1118
198394082 17 QLP 176 0 0 1126
198394082 18 QLP 153 0 0 1103
198394082 19 QLP 235 0 0 1150
198394082 20 QLP 61 0 0 1183
198394082 21 QLP 269 0 0 1262
198394082 22 QLP 153 0 0 1246
198394082 23 QLP 279 0 0 1235
198394082 24 SPOC 4279 0 0 18217
198394082 25 QLP 328 0 0 1167
198394082 26 QLP 251 0 0 1146
198394082 40 SPOC 0 0 0 19611
198394082 41 SPOC 0 0 0 18322
198394082 47 SPOC 2058 0 0 18862
198394082 48 SPOC 3804 0 0 18989

TIC   sector author #bit14 #bit15 #bit16 #cadence
302332574 14 QLP 218 0 0 1237
302332574 15 QLP 220 0 0 1190
302332574 21 QLP 164 0 0 1262
302332574 41 SPOC 0 0 0 18322
302332574 48 SPOC 2844 888 47 18989

TIC   sector author #bit14 #bit15 #bit16 #cadence
8885834 48 SPOC 2881 1490 0 18989

TIC   sector author #bit14 #bit15 #bit16 #cadence
103691996 14 SPOC 0 0 0 18424
103691996 20 QLP 248 0 0 1183
103691996 21 SPOC 779 0 0 18758
103691996 40 SPOC 0 0 0 19611
103691996 41 SPOC 0 0 0 18322
103691996 47 SPOC 2347 0 0 18862
103691996 48 SPOC 3189 616 0 18989
```

Script to produce the stats
```python
print("TIC   sector author #bit14 #bit15 #bit16 #cadence")
for lc in lc_coll: 
    print(lc.targetid, lc.sector, lc.author, len(lc.quality[(lc.quality & 4096) != 0]), len(lc.quality[(lc.quality & 16384) != 0]), len(lc.quality[(lc.quality & 32768) != 0]), len(lc.quality))
```

**Comment by barentsen** [comment]
Closed via #1218.
